delira/training/base_trainer.py

```diff
@@ -802,3 +853,14 @@ def _search_for_prev_state(path, extensions=None):
             return latest_state_path, latest_epoch
 
         return None, 1
+
+    def register_callback(self, callback: AbstractCallback):
+        has_all_attrs = True
+        for attr in ("iter", "epoch",):
```

Comment:
`iter` is already checked in super class, do we want to check it here again.

Reply:
No, we don't